src/test/java/net/sf/jabref/logic/formatter/bibtexfields/HtmlToLatexFormatterTest.java

```diff
@@ -45,7 +45,7 @@ public void testHTML() {
         assertEquals("{\\\"{a}}", formatter.format("&auml;"));
         assertEquals("{\\\"{a}}", formatter.format("&#228;"));
         assertEquals("{\\\"{a}}", formatter.format("&#xe4;"));
-        assertEquals("{$\\Epsilon$}", formatter.format("&Epsilon;"));
+        assertEquals("{{$\\Epsilon$}}", formatter.format("&Epsilon;"));
```

Comment:
Writing `\"a` is OK, but it is not wise for a piece of software to replace ä with `\"a` as there are many situations where it will lead to errors. So, if a single context independent string should be used it is `{\"{a}}` as that will always give the correct output. Same thing with `&Epsilon;`, if we want to guarantee that Ε is produced under all circumstances, we should use `{{$\Epsilon$}}`. Yes, kerning will not be perfect, but I'd rather take that than ε.


One should probably differ between two cases here. The Ägypten example is actually quite good. Somehow, I would think that this should be converted to {\"{A}}gypten as we cannot know that the Ä should always be capital (in the general case). On the other hand, in most cases, if we get &Delta; we can be quite sure that we actually want  Δ  as this is most likely used as a symbol rather than a character in a word. So, yes, somehow it is inconsistent.

Automatically adding {} will never work properly as *it is completely OK to have capital letters without {} in a .bib-file.* One should just be aware of the fact that the .bst determines the outcome. I'm therefore quite doubtful about that checker and will never add {} to any capital letters as long as I know for sure that they always should be capital as the .bst-file will take care of it (and that is the purpose of the .bst-file). That is the reason of the CaseKeeper using a list of words that should be kept in capital form. An improvement here would be to either extend these lists and/or support a user-defined list (that would solve the Ägypten case in a clearly better way).

If someone wants to write a better converter feel free to do so, but please make sure that it is indeed better and produces consistent code yielding correct output in the final document. The current one is far from perfect, but it does its job quite well when looking at the end result (it has saved me an immense amount of time if nothing else) and not merging these things in because someone might write a better converter at some stage seems quite stupid to me.

Again, feel free to update the internal format, but make sure you also add the HTML stuff. And one can wonder how often ✎ is really needed...